adapters/Pangle/PangleAdapter/Bidding/GADPangleRTBNativeRenderer.m

```diff
@@ -0,0 +1,201 @@
+    }
+    return nil;
+}
+
+- (NSString *)body {
+    if (_nativeAd && _nativeAd.data) {
+        return _nativeAd.data.AdDescription;
+    }
+    return nil;
+}
+
+- (NSString *)callToAction {
+    if (_nativeAd && _nativeAd.data) {
+        return _nativeAd.data.buttonText;
+    }
+    return nil;
+}
+
+- (NSDecimalNumber *)starRating {
+    return [NSDecimalNumber decimalNumberWithString:[NSString stringWithFormat:@"%ld",(long)_nativeAd.data.score]];
+}
+
+- (NSString *)store {
+    return nil;
+}
+
+- (NSString *)price {
+    return nil;
+}
+
+- (NSString *)advertiser {
+    if (_nativeAd && _nativeAd.data) {
+        return _nativeAd.data.AdTitle;
+    }
+    return nil;
+}
+
+- (NSDictionary<NSString *,id> *)extraAssets {
+    if (_nativeAd) {
+        return @{BUDNativeAdTranslateKey:_nativeAd};
+    }
+    return nil;
+}
+
+- (GADNativeAdImage *)imageWithUrlString:(NSString *)urlString {
+    NSURL *url = [NSURL URLWithString:urlString];
+    NSData *data = [NSData dataWithContentsOfURL:url];
+    UIImage *image = [UIImage imageWithData: data];
+    return [[GADNativeAdImage alloc] initWithImage:image];
+}
+
+#pragma mark BUNativeAdDelegate
+
+- (void)nativeAdDidLoad:(BUNativeAd *)nativeAd view:(UIView *_Nullable)view {
+    BUMaterialMeta *materialMeta = nativeAd.data;
+    // Set main image of the ad.
+    if (materialMeta.imageAry && materialMeta.imageAry.count && materialMeta.imageAry[0].imageURL != nil){
+        _images = @[[self imageWithUrlString:materialMeta.imageAry[0].imageURL]];
+    }
+    
+    // Set icon image of the ad.
+    if (materialMeta.icon && materialMeta.icon.imageURL != nil){
+        _icon = [self imageWithUrlString:_nativeAd.data.icon.imageURL];
+    }
+    
+    [_relatedView refreshData:nativeAd];
+    
+    if (_loadCompletionHandler) {
+        _delegate = _loadCompletionHandler(self,nil);
+    }
+}
+
+- (void)nativeAd:(BUNativeAd *)nativeAd didFailWithError:(NSError *_Nullable)error {
+    if (_loadCompletionHandler) {
+        _loadCompletionHandler(nil, error);
+    }
+}
+
+- (void)nativeAdDidBecomeVisible:(BUNativeAd *)nativeAd {
+    id<GADMediationNativeAdEventDelegate> delegate = _delegate;
+    [delegate reportImpression];
+}
+
+- (void)nativeAdDidClick:(BUNativeAd *)nativeAd withView:(UIView *_Nullable)view {
+    id<GADMediationNativeAdEventDelegate> delegate = _delegate;
+    [delegate reportClick];
+}
+
+- (void)nativeAd:(BUNativeAd *_Nullable)nativeAd adContainerViewDidRemoved:(UIView *)adContainerView {
+    
+}
+
+- (void)didRenderInView:(nonnull UIView *)view
+       clickableAssetViews:
+           (nonnull NSDictionary<GADNativeAssetIdentifier, UIView *> *)clickableAssetViews
+    nonclickableAssetViews:
+        (nonnull NSDictionary<GADNativeAssetIdentifier, UIView *> *)nonclickableAssetViews
+         viewController:(nonnull UIViewController *)viewController {
+    [_nativeAd registerContainer:view withClickableViews:clickableAssetViews.allValues];
+}
```

Comment:
-- BLOCKING--

- The ad attribution (i.e. the yellow "Ad" icon) and background of the native ad shouldn't be clickable.
- When the user clicks on Pangle's "AD" icon, they should be redirected to Pangle's privacy site.

Reply:
Any update on this?

Reply:
OK,I will fix it

Reply:
@wangyuanyuanyy This is a blocking issue, would you have any update on this?

Reply:
Just saw your comment. Thanks!

Reply:
Here is a bug in the Pangle SDK, SDK RD will be fixed this Wednesday. adapter does not need to be updated